modules/indexing/src/main/java/org/apache/ignite/internal/visor/query/VisorQueryDetailMetricsCollectorTask.java

```diff
@@ -116,21 +116,31 @@ private static void aggregateMetrics(long since, Map<GridCacheQueryDetailMetrics
         ) throws IgniteException {
             assert arg != null;
 
-            GridCacheProcessor cacheProc = ignite.context().cache();
-
-            Collection<String> cacheNames = cacheProc.cacheNames();
-
             Map<GridCacheQueryDetailMetricsKey, GridCacheQueryDetailMetricsAdapter> aggMetrics = new HashMap<>();
 
-            for (String cacheName : cacheNames) {
-                if (!isSystemCache(cacheName)) {
-                    IgniteInternalCache<Object, Object> cache = cacheProc.cache(cacheName);
-
-                    if (cache == null || !cache.context().started())
-                        continue;
-
-                    aggregateMetrics(arg.getSince(), aggMetrics, cache.context().queries().detailMetrics());
-                }
+            Collection<QueryHistoryMetrics> metrics = ((IgniteH2Indexing)ignite.context().query().getIndexing())
+                .runningQueryManager().queryHistoryMetrics().values();
+
+            for (QueryHistoryMetrics m: metrics) {
+                GridCacheQueryDetailMetricsKey key = new GridCacheQueryDetailMetricsKey(SQL_FIELDS, m.query());
+
+                GridCacheQueryDetailMetricsAdapter oldMetrics = aggMetrics.get(key);
+
+                GridCacheQueryDetailMetricsAdapter total = oldMetrics != null
+                    ? new GridCacheQueryDetailMetricsAdapter(SQL_FIELDS, m.query(), null,
+                        oldMetrics.executions() + (int)m.executions(),
+                        oldMetrics.completions() + (int)(m.executions() - m.failures()),
+                        oldMetrics.failures() + (int)m.failures(),
+                        Long.min(oldMetrics.minimumTime(), m.minimumTime()),
+                        Long.max(oldMetrics.maximumTime(), m.maximumTime()),
+                        oldMetrics.totalTime(),
+                        Long.max(oldMetrics.lastStartTime(), m.lastStartTime()),
+                        key)
+                    : new GridCacheQueryDetailMetricsAdapter(SQL_FIELDS, m.query(), null,
+                        (int)m.executions(), (int)(m.executions() - m.failures()), (int)m.failures(), m.minimumTime(),
```

Comment:
Please, refactor to one line per parameter. This GG code style.

Reply:
Refactored.